explorer/src/Pos/Explorer/Web/Server.hs

```diff
@@ -724,15 +751,16 @@ cAddrToAddr cAddr@(CAddress rawAddrText) =
             -- Originally taken from:
             -- * cardano-sl/tools/src/keygen/Avvm.hs
             -- * cardano-sl/tools/src/addr-convert/Main.hs
-            unless (BS.length addr == 32) $
-                throwM badAddressLength
-            pure $ makeRedeemAddress $ redeemPkBuild addr
+            if (BS.length addr /= 32)
+                then badAddressLength
+                else makeRedeemAddress $ redeemPkBuild addr
         Nothing ->
             -- cAddr is in Cardano address format
-            either badCardanoAddress pure (fromCAddress cAddr)
+            error "Undecoded address"
+            -- either badCardanoAddress const (fromCAddress cAddr)
   where
-    badAddressLength = Internal "Address length is not equal to 32, can't be redeeming pk"
-    badCardanoAddress = const $ throwM $ Internal "Invalid Cardano address!"
+    badAddressLength  = error "Address length is not equal to 32, can't be redeeming pk"
+    -- badCardanoAddress = error "Invalid Cardano address!"
```

Comment:
Why pure exceptions now? And this looks half-finished, things still commented out.

Reply:
Pure exceptions so I can test them quickly. Commented out the things that are not interesting to us.